clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/providers/TitusClusterProvider.groovy

```diff
@@ -195,6 +213,20 @@ class TitusClusterProvider implements ClusterProvider<TitusCluster>, ServerGroup
     serverGroup
   }
 
+  private List<Map<String, String>> getTaggedImageAttributes(String account, Job job) {
```

Comment:
This method signature could use some love. Either [L197](https://github.com/spinnaker/clouddriver/pull/5035/files#diff-8898a6601d27549d1304e49f079d5827142b15e673ddd34794461efee1482460R197) isn't correctly handling this method's return type and has bugs, or this method is returning the wrong type.

Inferring what I can from your current usage, this method returns either 0 or 1 list elements. If that is the case, then this method should probably return `Optional<Map<String, String>>` so you could do:

```
serverGroup.buildInfo.jenkins = getTaggedImage.ifPresentOrElse(
  { it -> 
    [
      "name": it.jenkinsJob ?: "NA", 
      "number": it.buildNumber ?: "NA", 
      "host": it.jenkinsHost ?: "NA"
    ] 
  },
  { [:] }
)
```

Doing so would make the method usage not only much clearer, but less prone to bugs.

Reply:
👍 made some updates based on your input.

Reply:
Did you test this? You changed the signature, but didn't change the method body at all.

Reply:
Yes , and it seems to work fine , do you see any other concerns ?